velox/exec/tests/RowContainerTest.cpp

```diff
@@ -729,6 +729,26 @@ TEST_F(RowContainerTest, rowSizeWithNormalizedKey) {
   ASSERT_EQ(numRows, 1);
 }
 
+TEST_F(RowContainerTest, estimateRowSize) {
+  auto data = makeRowContainer({BIGINT()}, {VARCHAR()});
+  auto unlimited = 100'000;
+  for (auto i = 0; i < unlimited; i++) {
+    data->newRow();
+  }
+
+  RowContainerIterator iter;
+  char* result[unlimited];
+  // Batch sizes smaller than the total row container size.
+  std::vector<int64_t> batchSizes = {1, 1 << 10, 1 << 20};
+  for (auto batchSize : batchSizes) {
+    // Get the expected numRows based on the given batchSize.
+    auto numRows = data->listRows(&iter, unlimited, batchSize, result);
+    // Test estimateRowSize is accurate to get the expected numRows.
+    EXPECT_EQ(
+        std::ceil(batchSize * 1.0 / data->estimateRowSize().value()), numRows);
```

Comment:
I still don't understand how this test verifies the behavior of data->estimateRowSize(). This behavior should not depend on the 'batchSize', hence, I don't see why logic around 'batchSizes' is relevant. At the same time, I'd expect this test to check scenarios where (1) data->estimateRowSize() returns std::nullopt; (2) data is variable-width and data->estimateRowSize() is taking that part of the data into account.